oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderTests.java

```diff
@@ -346,6 +347,23 @@ public void decodeWhenSignatureMismatchesAlgorithmThenThrowsException() throws E
 				.isInstanceOf(BadJwtException.class);
 	}
 
+	// gh-8730
+	@Test
+	public void decodeWhenCustomizingJwtProcessor() throws Exception {
```

Comment:
While I agree that this test name makes sense, you might consider naming it to something that reflects the use case you are wanting the functionality to address, e.g. `decodeWhenCustomTypHeaderThenProcesses` or something like that.